File: 4-Website/LC14/Main.java
    public String ans(String[] strs) {
        
        String c = "";
        int lenSmallestWord = Integer.MAX_VALUE;
        
        for (String word: strs){
            lenSmallestWord = Math.min(lenSmallestWord, word.length());
        }
        
        for (int i = 0; i < lenSmallestWord; i++) {
            String first = strs[0];
            char letter = first.charAt(i);
            
            for (String word: strs) {
                if (word.charAt(i) != letter) {
                    return first.substring(0, i);
                }
            }
            
            c += letter;
        }
        
        return c;
    }

File: 4-Website/LC14/main.py
def p1(strs):
    
    common = ''
    
    smallest_word_len = sys.maxsize
    
    for word in strs:
        smallest_word_len = min(len(word), smallest_word_len)
    
    for i in range(smallest_word_len):
        first_word = strs[0]
        letter = first_word[i]
        
        for word in strs:
            if word[i] != letter:
                return first_word[:i]
        
        common += letter
    
    return common
